### PR TITLE
Create shiny sass variable default files specific to Bootstrap Version

### PR DESCRIPTION
**Pull request by gadenbuie**
I reverted `inst/www/shared/shiny_scss/bootstrap.scss` back to its state before the BS 5.3 updates. Those updates tried to support BS 3, 4 and 5.3 in the same `bootstrap.scss` file.

Then I created variants for each major version of Bootstrap. This allows us to confidently use features of each version of Bootstrap without a lot of checking for variable existence or having to backport Sass variables for older versions of Bootstrap.

The original `bootstrap.scss` had a small amount of backporting to simultaneously support BS3 and BS4; I removed the backports in each version.

### CODE REVIEW

File: inst/www/shared/shiny_scss/shiny.bootstrap4.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+$notification-error-border:         $border-width solid mix($bg, map-get($theme-colors, "danger"), 79%) !default;
+$notification-close-color:          mix($bg, $fg, 20%) !default;
+$notification-close-hover-color:    $fg !default;
+$notification-content-action-color: map-get($theme-colors, "primary") !default;
+$datepicker-disabled-color:         $dropdown-link-disabled-color !default;
+
+$shiny-file-active-shadow: $custom-file-focus-box-shadow !default;
+
+// TODO: should this be changed to danger?
+//$shiny-error-color:                 red !default;
```
Comment (cpsievert) [inline inst/www/shared/shiny_scss/shiny.bootstrap4.scss:34]: Can keep if you want, but I'd be fine removing

```suggestion
```

File: inst/www/shared/shiny_scss/shiny.bootstrap4.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
+$bg: $body-bg !default;
+$fg: $body-color !default;
+
+$shiny-disconnected-bg:             mix($bg, $fg, 58%) !default;
+$shiny-table-na:                    mix($bg, $fg, 54%) !default;
+$shiny-error-validated-color:       mix($bg, $fg, 50%) !default;
+// TODO: what should this be? mix of primary and bg?
```
Comment (cpsievert) [inline inst/www/shared/shiny_scss/shiny.bootstrap4.scss:7]: Can keep if you want, but I'd be fine removing

```suggestion
```

File: inst/www/shared/shiny_scss/shiny.bootstrap3.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+// No $border-width, $border-color, or $border-radius in BS3
+$border-width: 1px !default;
+$border-color: mix($bg, $fg, 70%) !default;
+$border-radius: $border-radius-base !default;
+
+$shiny-disconnected-bg:             mix($bg, $fg, 58%) !default;
+$shiny-table-na:                    mix($bg, $fg, 54%) !default;
+$shiny-error-validated-color:       mix($bg, $fg, 50%) !default;
+// TODO: what should this be? mix of primary and bg?
```
Comment (cpsievert) [inline inst/www/shared/shiny_scss/shiny.bootstrap3.scss:13]: Can keep if you want, but I'd be fine removing

```suggestion
```

**Review by cpsievert — APPROVED** [review]
LGTM, thanks. Would also be great to have a short explanation of what this PR changes in terms of default Bootstrap 5 styles. Please also update news.

**Comment by cpsievert** [comment]
@gadenbuie it'd be great to get this in fairly soon, since it seems like it could impact testing screenshots.

To that end, I am a bit surprised that `inst/www/shared/shiny.min.css` hasn't changed as a result of these changes, so before merging, let's make sure that is updated (if need be)

**Comment by gadenbuie** [comment]
> To that end, I am a bit surprised that `inst/www/shared/shiny.min.css` hasn't changed as a result of these changes, so before merging, let's make sure that is updated (if need be)

Is that file compiled against BS3? In which case it'd be a good thing we don't see changes.

**Comment by gadenbuie** [comment]
@cpsievert Sorry, I didn't realize this was waiting on changes from me. I've removed the comments and added a news item.

**Comment by cpsievert** [comment]
> Is that file compiled against BS3? In which case it'd be a good thing we don't see changes.

Ah, good point!

**Pull request merged**